src/actions/guides/http_and_routing/routing_and_params.cr

```diff
@@ -211,7 +211,31 @@ class Guides::HttpAndRouting::RoutingAndParams < GuideAction
     * Multiple namespaces: `Api::V1::Projects::Users::Show`  -> `get "/api/v1/projects/:project_id/users/:user_id"`
     * Multi-word namespace: `MyAdminSection::Projects::Users::Show`  -> `get "/my_admin_section/projects/:project_id/users/:user_id"`
 
-    ## Catch-all and 404 errors
+    ## Fallback routing
+
+    For some apps you may want a wildcard/catch-all behavior instead of rendering
+    some HTML when Lucky can't find a route. For example, this type of behavior
+    can be useful for Single Page Applications (SPAs) so that you can handle
+    routing client-side.
+
+    To do this, use the `fallback` macro.
+
+    ```crystal
+    # in src/actions/frontend/index.cr
+    class Frontend::Index < BrowserAction
+      fallback do
+        if html?
+          render Home::IndexPage
+        else
+          raise Lucky::RouteNotFoundError.new(context)
+        end
+      end
+    end
+    ```
+
+    > ProTip: The `fallback` should always contain a `Lucky::RouteNotFoundError` error. This is to throw a 404 when an asset, or some other file is not found.
```

Comment:
We haven't used `ProTip` elsewhere so I'm nervous about introducing that. I think we can remove it and the visuals should show it is a tip because it looks different